UpsertedProfileOrchestrator/handler.ts

```diff
@@ -38,201 +38,214 @@ export const OrchestratorInput = t.intersection([
-    yield context.df.callActivityWithRetry(
-      "SendWelcomeMessagesActivity",
-      retryOptions,
-      {
-        messageKind: "HOWTO",
-        profile: newProfile
-      } as SendWelcomeMessageActivityInput
-    );
-  }
 
-  // Update subscriptions feed
-  if (profileOperation === "CREATED") {
-    // When a profile get created we add an entry to the profile subscriptions
+    // Send welcome messages to the user
+    const isInboxEnabled = newProfile.isInboxEnabled === true;
+    const hasOldProfileWithInboxDisabled =
+      profileOperation === "UPDATED" && oldProfile.isInboxEnabled === false;
+
+    const hasJustEnabledInbox =
+      isInboxEnabled &&
+      (profileOperation === "CREATED" || hasOldProfileWithInboxDisabled);
+
     context.log.verbose(
-      `${logPrefix}|Calling UpdateSubscriptionsFeedActivity|OPERATION=SUBSCRIBED`
-    );
-    yield context.df.callActivityWithRetry(
-      "UpdateSubscriptionsFeedActivity",
-      retryOptions,
-      {
-        fiscalCode: newProfile.fiscalCode,
-        operation: "SUBSCRIBED",
-        subscriptionKind: "PROFILE",
-        updatedAt: updatedAt.getTime(),
-        version: newProfile.version
-      } as UpdateServiceSubscriptionFeedActivityInput
-    );
-  } else {
-    // When the profile gets updates, we extract the services that have been
-    // blocked and unblocked during this profile update.
-    // Blocked services get mapped to unsubscribe events, while unblocked ones
-    // get mapped to subscribe events.
-    const {
-      e1: unsubscribedServices,
-      e2: subscribedServices
-    } = diffBlockedServices(
-      oldProfile.blockedInboxOrChannels,
-      newProfile.blockedInboxOrChannels
+      `${logPrefix}|OPERATION=${profileOperation}|INBOX_ENABLED=${isInboxEnabled}|INBOX_JUST_ENABLED=${hasJustEnabledInbox}`
     );
 
-    for (const s of subscribedServices) {
-      context.log.verbose(
-        `${logPrefix}|Calling UpdateSubscriptionsFeedActivity|OPERATION=SUBSCRIBED|SERVICE_ID=${s}`
+    if (hasJustEnabledInbox) {
+      yield context.df.callActivityWithRetry(
+        "SendWelcomeMessagesActivity",
+        retryOptions,
+        {
+          messageKind: "WELCOME",
+          profile: newProfile
+        } as SendWelcomeMessageActivityInput
       );
       yield context.df.callActivityWithRetry(
-        "UpdateSubscriptionsFeedActivity",
+        "SendWelcomeMessagesActivity",
         retryOptions,
         {
-          fiscalCode: newProfile.fiscalCode,
-          operation: "SUBSCRIBED",
-          serviceId: s as ServiceId,
-          subscriptionKind: "SERVICE",
-          updatedAt: updatedAt.getTime(),
-          version: newProfile.version
-        } as UpdateServiceSubscriptionFeedActivityInput
+          messageKind: "HOWTO",
+          profile: newProfile
+        } as SendWelcomeMessageActivityInput
       );
+      if (params.sendCashbackMessage) {
+        yield context.df.callActivityWithRetry(
+          "SendWelcomeMessagesActivity",
```

Comment:
Maybe we have to rename `SendWelcomeMessagesActivity`.

Reply:
Why? They are always welcome messages sent on user profile creation. What name do you suggest?